torch/_inductor/codegen/wrapper.py

```diff
@@ -1161,6 +1162,16 @@ def make_buffer_allocation(self, buffer):
         return self.make_allocation(buffer.get_name(), device, dtype, shape, stride)
 
     def make_allocation(self, name, device, dtype, shape, stride):
+        if device.type == "cpu" and len(shape) <= 8:
+            # optimized path for faster allocations, saving ~2us versus the stuff below
+            # TODO(jansel): can we do similar things on the GPU?
```

Comment:
I also wrote `empty_strided_cuda` in `ATen/cuda/EmptyTensor.h` though you would have to deal with cpu-only builds where it may not be available. Possibly through a dispatch stub, or `ifdef USE_CUDA` may be enough.

You could also use `empty_cpu` and `empty_cuda` for the contiguous cases. This should be slightly faster still as they are able to do fast `TensorImpl` restriding, though after python overhead I'm not sure how much that will amount to.

Reply:
Next PR in the stack will add `empty_strided_cuda`.